crypto/bio/bio_addr.c

```diff
@@ -672,6 +672,7 @@ int BIO_lookup_ex(const char *host, const char *service, int lookup_type,
                   int family, int socktype, int protocol, BIO_ADDRINFO **res)
 {
     int ret = 0;                 /* Assume failure */
+    int lock_err = 0;
```

Comment:
Instead of adding a variable here, I think you can just return 0 if the write_lock fails, as the only thing that happens at the err: label is the unlocking of this mutex.  This saves you the additional stack space and lets you avoid having to set and check it.

Additionally, the failure case for the RUN_ONCE call right above the write_lock also needs to just return 0, as the jump to the err: label there would unlock an already unlocked mutex